### DISCUSSION TITLE
How to change provider settings as easy as possible?

### DISCUSSION THREAD

**Discussion by ghost** [Josh-XT/AGiXT#955] [Q&A]
I want to change for example AGiXT/agixt/providers/kobold.py
But changes to the parameters of this file do not result in changes inside the docker containers.
I tried using 
docker-compose up -d --build 
docker-compose build --no-cache 
docker-compose up -d 
but that didn't work.
So far it has only worked like this:
 edit the kobold.py file
 **_docker cp (path to edited kobold.py file) (agixt-agixt-1 container id):(path to copy kobold.py file to agixt-agixt-1 container)_**
 Example of my command:
 **_docker cp /home/agi/kobold.py 5aff64bbcd29bb4ccacd868459078f931324eb53:/agixt/providers/kobold.py_**

```
import requests

class KoboldProvider:
    def __init__(
        self,
        AI_PROVIDER_URI: str = "",
        MAX_TOKENS: int = 16384,
        AI_TEMPERATURE: float = 0.5,
        AI_MODEL: str = "default",
        **kwargs,
    ):
        self.requirements = []
        self.AI_PROVIDER_URI = AI_PROVIDER_URI
        self.MAX_TOKENS = MAX_TOKENS if MAX_TOKENS else 16384
        self.AI_TEMPERATURE = AI_TEMPERATURE if AI_TEMPERATURE else 0.5
        self.AI_MODEL = AI_MODEL

    async def instruct(self, prompt, tokens: int = 0):
        try:
            max_tokens = int(self.MAX_TOKENS - tokens)
        except:
            max_tokens = 16384
        response = requests.post(
            f"{self.AI_PROVIDER_URI}/api/v1/generate",
            json={
                "prompt": prompt,
                "max_context_length": max_tokens,
                "max_length": 2000,
                "temperature": float(self.AI_TEMPERATURE),
            },
        )
        try:
            return response.json()["results"][0]["text"].replace("\n", "\n")
        except:
            return response.json()["detail"][0]["msg"].replace("\n", "\n")

```

**Comment by Josh-XT** [comment]
Didn't realize these discussions were here separately from issues. You can modify agent settings in the Streamlit web UI.
